### PR TITLE
test reader transpiler

### PR DESCRIPTION
**Pull request by JiayiFeng**

### CODE REVIEW

File: python/paddle/fluid/io.py [hunk 1]
```diff
@@ -744,3 +744,101 @@ def has_success(checkpoint_dir, cur_dir):
...
     return current_dir
+
+
+def get_test_program(filelist, program=None, startup_program=None):
+    """
+    Transpile current train program to a program to read test dataset
+    if the program is using reader ops like "open_files_op".
+    """
+
+    def _copy_reader_var_(block, var, new_name=None):
```
Comment (typhoonzero) [inline python/paddle/fluid/io.py:755]: Curious why we can not reuse this function?

File: python/paddle/fluid/io.py [hunk 2]
```diff
@@ -744,3 +744,101 @@ def has_success(checkpoint_dir, cur_dir):
...
+        if new_name == None:
+            new_name = var.name
+        new_var = block.create_var(
+            name=str(new_name), type=core.VarDesc.VarType.READER)
+        new_var.desc.set_shapes(var.desc.shapes())
+        new_var.desc.set_dtypes(var.desc.dtypes())
+        new_var.persistable = True
+        return new_var
+
+    def get_test_reader_name(train_reader_name):
```
Comment (typhoonzero) [inline python/paddle/fluid/io.py]: Private functions should start with "_":

File: python/paddle/fluid/io.py (hunk 1 above)
Comment (JiayiFeng) [reply python/paddle/fluid/io.py:755]: The original `_copy_reader_var_ ` is in another module `fluid.layers.io`. To reuse it, we have to expose it. However, it seems strange to expose such an internal function and make it user accessible.

File: python/paddle/fluid/io.py (hunk 2 above)
Comment (JiayiFeng) [reply python/paddle/fluid/io.py]: Good. I will fix this.

File: python/paddle/fluid/io.py (hunk 1 above)
Comment (typhoonzero) [reply python/paddle/fluid/io.py:755]: I see.

**Review by typhoonzero — APPROVED** [review]
This LGTM now, I'll need to test it with my case, and may take some time. Thanks for refining this code!

**Comment by JiayiFeng** [comment]
@typhoonzero Thanks. Please let me know whether the test results are all ok.

**Pull request merged**